frontend/src/reusable/BasicForm.tsx

```diff
@@ -0,0 +1,98 @@
+import { useForm } from "@mantine/form";
+import { Button, Fieldset, Group } from "@mantine/core";
+import { PromptDefinition } from "./PromptTypes";
+import { ReactNode } from "react";
+
+type PromptCollectorProps = { promptDefinitions: PromptDefinition[] };
+
+function updateImageBlur(
+    imageName: string,
+    blurQuantity: number = 9,
+    blurScaling: number = 7,
+) {
+    const image = document.getElementById(imageName);
+    if (!image) {
+        console.error(`Cannot find image with name ${imageName}`);
+    } else {
+        image?.classList.remove("blur-3xl");
```

Comment:
i'm not a huge fan of this because if we change the default blur, we have to do it in two different places now. was there a problem with calling updateImageBlur once before the component mounts?

Reply:
yes because this isnt a component, its a function so its technically a node. this is just what i remember from what i was working on during this. i tried to recreate componentDidMount but its not really possible in this case so your solution didnt work anymore. i know its an annoying work-around but it works.